packages/clerk-js/src/ui/hooks/useFetch.ts

```diff
@@ -24,11 +24,11 @@ export const useFetch = <T>(
     requestStatus.setLoading();
     fetcherRef
       .current(params)
-      .then(domain => {
+      .then(result => {
         requestStatus.setIdle();
-        if (domain) {
-          setData({ ...domain });
-          callbacks?.onSuccess?.({ ...domain });
+        if (typeof result !== 'undefined') {
```

Comment:
Note for reviewer: if value is an object then do a shallow copy otherwise use the actual value